engine/table/src/main/java/io/deephaven/engine/util/PythonScope.java

```diff
@@ -143,7 +143,12 @@ default Collection<String> getKeysCollection() {
     /**
      * @return the Python's __main__ module namespace
      */
-    public PyDictWrapper mainGlobals();
+    PyDictWrapper mainGlobals();
+
+    /**
+     * @return the current scope or the main globals if no scope is set
+     */
+    PyDictWrapper currentScope();
```

Comment:
Why do we need this in the interface?

Reply:
Previously we were using `getKeysRaw` (through delegation) and `getEntriesRaw` both of which created a `Stream` around iterating items in the `currentScope`. Now we must make a copy/dictCopy prior to using the returned values. (We were assuming that we could `ensureGil` while collecting keys and/or values, but this is not the case as performing these operations involve JNI.

I will remove the superfluous methods since I don't believe there is a good way to use them safely anymore.